### PR TITLE
pkg: added multinetwork ingress, egress rules

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/networkpolicy/multinetegressrule.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+	return builder
+}
+
+// WithPortAndProtocol adds port and protocol to Egress rule.
+func (builder *EgressRuleBuilder) WithPortAndProtocol(port uint16, protocol v1.Protocol) *EgressRuleBuilder {
+	glog.V(100).Infof("Adding port %d and protocol %s to EgressRule", port, protocol)
+
+	if port == 0 {
+		glog.V(100).Infof("Invalid port number can not be 0")
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetegressrule.go:42]: Port number can not be 0

File: pkg/networkpolicy/multinetegressrule.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+		builder.errorMsg = "Invalid port number can not be 0"
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetegressrule.go:44]: Port number can not be 0

File: pkg/networkpolicy/multinetegressrule.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+			}
+		}
+	}
+
+	return builder
+}
+
+// WithPeerPodSelector adds pod selector to Egress rule.
+func (builder *EgressRuleBuilder) WithPeerPodSelector(podSelector metaV1.LabelSelector) *EgressRuleBuilder {
+	glog.V(100).Infof("Adding peer podselector %v to EgressRule", podSelector)
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetegressrule.go:82]: pod selector

File: pkg/networkpolicy/multinetegressrule.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+	builder.definition.To = append(builder.definition.To, v1beta1.MultiNetworkPolicyPeer{PodSelector: &podSelector})
+
+	return builder
+}
+
+// WithPeerPodSelectorAndCIDR adds pod selector and CIRD to Egress rule.
+func (builder *EgressRuleBuilder) WithPeerPodSelectorAndCIDR(
+	podSelector metaV1.LabelSelector, cidr string, except ...[]string) *EgressRuleBuilder {
+	glog.V(100).Infof("Adding peer podselector %v to EgressRule", podSelector)
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetegressrule.go:96]: pod selector

File: pkg/networkpolicy/multinetegressrule.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+	_, _, err := net.ParseCIDR(cidr)
+
+	if err != nil {
+		glog.V(100).Infof("Invalid CIRD %s", cidr)
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetegressrule.go:101]: CIDR

File: pkg/networkpolicy/multinetegressrule.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+		builder.errorMsg = fmt.Sprintf("Invalid CIDR argumetn %s", cidr)
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetegressrule.go:103]: argument

File: pkg/networkpolicy/multinetegressrule.go [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.WithPeerPodSelector(podSelector)
+
+	// Append IPBlock config to the previosly added rule
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetegressrule.go:112]: previously

File: pkg/networkpolicy/multinetingressrule.go [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+	glog.V(100).Infof("Initializing new Ingress rule structure")
+
+	builder := &IngressRuleBuilder{
+		definition: &v1beta1.MultiNetworkPolicyIngressRule{},
+	}
+
+	return builder
+}
+
+// WithPortAndProtocol adds port and protocol and protocol to Ingress rule.
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetingressrule.go:36]: extra "and protocol"

File: pkg/networkpolicy/multinetingressrule.go [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+func (builder *IngressRuleBuilder) WithPortAndProtocol(port uint16, protocol v1.Protocol) *IngressRuleBuilder {
+	glog.V(100).Infof("Adding port %d protocol %s to IngressRule", port, protocol)
+
+	if port == 0 {
+		glog.V(100).Infof("Invalid port number can not be 0")
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetingressrule.go:41]: strip the Invalid

File: pkg/networkpolicy/multinetingressrule.go [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+
+		builder.errorMsg = "Invalid port number can not be 0"
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetingressrule.go:43]: strip the Invalid

File: pkg/networkpolicy/multinetingressrule.go [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+			}
+		}
+	}
+
+	return builder
+}
+
+// WithPeerPodSelector adds port and protocol to Ingress rule.
+func (builder *IngressRuleBuilder) WithPeerPodSelector(podSelector metaV1.LabelSelector) *IngressRuleBuilder {
+	glog.V(100).Infof("Adding peer podselector %v to Ingress Rule", podSelector)
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetingressrule.go:81]: Adding port and protocol

File: pkg/networkpolicy/multinetingressrule.go [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+
+	builder.definition.From = append(
+		builder.definition.From, v1beta1.MultiNetworkPolicyPeer{
+			PodSelector: &podSelector,
+		})
+
+	return builder
+}
+
+// WithCIDR adds CIRD to Ingress rule.
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetingressrule.go:95]: adds CIDR

File: pkg/networkpolicy/multinetingressrule.go [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+func (builder *IngressRuleBuilder) WithCIRD(cidr string, except ...[]string) *IngressRuleBuilder {
+	glog.V(100).Infof("Adding peer CIDR %s to Ingress Rule", cidr)
+
+	_, _, err := net.ParseCIDR(cidr)
+
+	if err != nil {
+		glog.V(100).Infof("Invalid CIRD %s", cidr)
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetingressrule.go:102]: CIDR

File: pkg/networkpolicy/multinetingressrule.go [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+
+		builder.errorMsg = fmt.Sprintf("Invalid CIDR argumetn %s", cidr)
+	}
+
+	if len(except) > 0 {
+		glog.V(100).Infof("Adding CIRD except %s to Ingress Rule", except[0])
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetingressrule.go:108]: CIDR

File: pkg/networkpolicy/multinetingressrule.go [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	if builder.definition.From == nil {
+		builder.definition.From = append(builder.definition.From, v1beta1.MultiNetworkPolicyPeer{})
+	}
+
+	// Append IPBlock config to the previosly added rule
```
Comment (achuzhoy) [inline pkg/networkpolicy/multinetingressrule.go:119]: previously

**Review by cdvultur — COMMENTED** [review]
Overall looks good. Only the spelling requires some changes

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**